com.play0ad.zeroad.json

```diff
@@ -1,120 +1,113 @@
-				}
-			]
-		}
-	]
+  "app-id": "com.play0ad.zeroad",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "1.6",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "0ad",
+  "finish-args": [
+    "--socket=wayland",
+    "--socket=x11",
+    "--socket=pulseaudio",
+    "--share=network",
+    "--device=dri"
+  ],
+  "build-options": {
+    "cflags": "-O2",
+    "cxxflags": "-O2"
+  },
+  "cleanup": [
+    "/include",
+    "/lib/pkgconfig",
+    "*.a",
+    "*.la"
+  ],
+  "modules": [
+    {
+      "name": "boost",
+      "buildsystem": "simple",
+      "build-commands": [
+        "./bootstrap.sh --prefix=/app --with-libraries=filesystem",
+        "./b2 headers",
+        "./b2 install"
+      ],
+      "sources": [
+        {
+          "sha256": "2684c972994ee57fc5632e03bf044746f6eb45d4920c343937a465fd67a5adba",
+          "type": "archive",
+          "url": "https://dl.bintray.com/boostorg/release/1.67.0/source/boost_1_67_0.tar.bz2"
+        }
+      ]
+    },
+    {
+      "name": "gloox",
+      "sources": [
+        {
+          "sha256": "0243086c0f4f0440d6d8e55705f83249a4463a1d75a034be42b5312e8886dea8",
+          "type": "archive",
+          "url": "https://camaya.net/download/gloox-1.0.20.tar.bz2"
+        }
+      ]
+    },
+    {
+      "name": "enet",
+      "sources": [
+        {
+          "sha256": "e36072021faa28731b08c15b1c3b5b91b911baf5f6abcc7fe4a6d425abada35c",
+          "type": "archive",
+          "url": "http://enet.bespin.org/download/enet-1.3.13.tar.gz"
+        }
+      ]
+    },
+    {
+      "name": "miniupnpc",
+      "buildsystem": "cmake-ninja",
+      "sources": [
+        {
+          "sha256": "e19fb5e01ea5a707e2a8cb96f537fbd9f3a913d53d804a3265e3aeab3d2064c6",
+          "type": "archive",
+          "url": "https://miniupnp.tuxfamily.org/files/miniupnpc-2.1.tar.gz"
+        }
+      ]
+    },
+    {
+      "name": "libsodium",
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://download.libsodium.org/libsodium/releases/libsodium-1.0.16.tar.gz",
+          "sha256": "eeadc7e1e1bcef09680fb4837d448fbdf57224978f865ac1c16745868fbd0533"
+        }
+      ]
+    },
+    {
+      "name": "0ad",
+      "buildsystem": "simple",
+      "build-commands": [
+        "cd build/workspaces && ./update-workspaces.sh -j$FLATPAK_BUILDER_N_JOBS --disable-atlas --bindir=/app/bin --libdir=/app/lib --datadir=/app/share/games/0ad/ && cd gcc && make -j$FLATPAK_BUILDER_N_JOBS"
+      ],
+      "post-install": [
+        "mkdir --parents /app/share/games/0ad /app/lib/games/0ad app/games",
+        "mv binaries/data/l10n /app/share/games/0ad/",
+        "mv binaries/system/pyrogenesis /app/bin/",
+        "mv binaries/system/lib* /app/lib/",
+        "install -Dm644 build/resources/0ad.appdata.xml /app/share/metainfo/com.play0ad.zeroad.appdata.xml",
+        "install -Dm644 build/resources/0ad.desktop /app/share/applications/com.play0ad.zeroad.desktop",
```

Comment:
There is no icon in the installed app, because the name doesn't match. You have to add:
`"desktop-file-edit --set-key=Icon --set-value='com.play0ad.zeroad' /app/share/applications/com.play0ad.zeroad.desktop"`